### ISSUE TITLE
No hits below 94% similarity

### ISSUE THREAD

**Issue by vanessamata** [DominikBuchner/BOLDigger3#5]
Hi Dominik!

Thanks for quickly updating BOLDigger to support BOLD V5!

I've noticed that now there seem to be a lot of "no matches" and when checking the hits there's no hits below 94% similarity, despite having selected database 3 (animal library, public + private) and mode 3 (Exhaustive Search). This issue seems to happen on the website as well, so I guess it is not a problem of BOLDigger itself, but rather some kind of bug with their identification engine. I have no idea when this will be fixed (i have already emailed them), so I wonder if BOLDigger (perhaps v2?) could keep accessing BOLD V4 instead?  I'm trying to identify things from Africa, so a lot of the hits are below 94%. I managed to get hits when going to BOLD V4 with 90-93% similarity, but get "no match" on V5.

Thanks!!

Vanessa

ps: right now the identification engine of V4 seems to be down though... at least I haven't been able to get any result.

**Comment by DominikBuchner** [comment]
Hi Vanessa,
the BOLD V4 api is also down, so without major updates to the code this is not easily doable. It might be possible to unlock different search parameters with some computer magic, I'll take a look.
Until then our best bet is to contact BOLD regarding the issue. I believe it should be an easy fix for them, and also believe mode 3 and database 3 should at least go down to 85% as they state on the website.

**Comment by vanessamata** [comment]
I got a reply saying they would look into it and get back to me as soon as possible, and 20 minutes later they asked for an example of a no match and of something that would only report hits above 94%. I have provided example sequences this morning and now I am waiting for feedback :)  fingers crossed that they solve the issue! :)

**Comment by DominikBuchner** [comment]
Perfect, thank you very much. When they fix it, BOLDigger3 will automatically adjust!

**Comment by Anto007** [comment]
Hi @DominikBuchner 

I've got a somewhat similar issue as reported here. Below are the results from BOLDigger3 for my COI ASVs
![image](https://github.com/user-attachments/assets/faf15f71-a48d-4b27-ae3d-e6f01f7c1847)
and below are the results from BOLDigger2 (that was generated months ago) for these same COI ASVs
![image](https://github.com/user-attachments/assets/581a24a7-95f7-4d09-ada6-cf7b79663ee3)
Obviously, the number of "No matches" is high in the results from BOLDigger3 presumably due to the 85% similarity cutoff in mode 3 in BOLD v5. In BOLDigger2, different thresholds (97%: species level, 95%: genus level, 90%: family level, 85%: order level, <85% and >= 50: class level) for the taxonomic levels were used to find the best fitting hit but I guess going up to 50% identity to get class-level classifications is not going to be possible anymore when using BOLD v5 database?

**Comment by DominikBuchner** [comment]
Yes and I would not trust anything below 85% anyways, therefore I removed the lower threshold entirely. I think this is actually a good improvement by BOLD if only it was working.

**Comment by Anto007** [comment]
Thanks for your response @DominikBuchner and I understand your point. However, a consequence of this will now be that a major chunk of our sequences from eDNA sequencing of poorly characterized environments is going to be reported as "Unclassified". I suppose the choice to go for a conservative approach or a liberal approach with respect to assigning taxonomy might be considered somewhat subjective and context-dependent.

**Comment by Anto007** [comment]
But my BOLDigger3 results above are from this morning and they have clearly taken hits > 85%. I didn't see a <94% problem at least in my results

**Comment by DominikBuchner** [comment]
Can you send the file with all results to my working mail address? I'd like to have a look, because I was able to reproduce the issue described!

**Comment by Anto007** [comment]
Sorry, I'm unable to follow you. You mean to say you too got results files that did not have results <94%? My results sheet from this morning has got plenty of hits > 85% and I had used boldigger3-1.1.2 (which I've upgraded further to 3-1.1.4 at this very minute but I'm yet to test this upgraded version)

**Comment by DominikBuchner** [comment]
So you get results between 0.85% and 0.94%? That would mean that they fixed
it immediately

Jant007 ***@***.***> schrieb am Do., 31. Okt. 2024, 14:56:

**Comment by vanessamata** [comment]
perhaps they fixed it already? i will try again to see if is solved!

**Comment by Anto007** [comment]
Yes, I didn't notice a <94% identity problem in my BOLDigger3 results from this morning's run (for example, see the 88.854% identity result for my ASV_9 in my earlier posted BOLDigger3 screenshot in this thread). I didn't get a chance to run BOLDigger3 yet again this day and so don't know if something broke at the BOLD server end for example later in the afternoon.

**Comment by DominikBuchner** [comment]
Hi all,
short update on that matter:

- The bug is not fixed yet and I can reproduce it via the website. I cannot get any pct_id value between 0.85 and 0.94
- @Anto007: It would be great if you could share the fasta you posted above with me so I can investigate this further because I did not find a single case with over 10k sequences. It does not seem to be a bug in BOLDigger but the ID engine itself
- Upon further investigation, it turns out that, when you know how to, you can freely choose the search parameters, e.g. reduce the pct_id to 70%, get 500 hits for each sequence, etc. I don't believe BOLD intends this so I won't implement it for now but contacted them to report the bug. 

TLDR: No hits below 0.94% for now, I'm in contact with BOLD.

**Comment by DominikBuchner** [comment]
Update: BOLD is aware of the problem and actively working to fix it.

**Comment by Anto007** [comment]
@DominikBuchner That's strange- perhaps, I got lucky then. Here are my input ASVs fasta file and the identification results that I got from BOLDigger3. 
[Test_ASVs_identification_result.xlsx](https://github.com/user-attachments/files/17600322/Test_ASVs_identification_result.xlsx)
[Test_ASVs.fa.txt](https://github.com/user-attachments/files/17600324/Test_ASVs.fa.txt)
There are 7033 ASV sequences in total and you'll notice that many of them indeed are <90% identity hits. You seem to have closed this issue but has the BOLD admins communicated to you that that they have fixed this issue?

**Comment by DominikBuchner** [comment]
Not fixed yet, but it seems to be a minor issue. What was the operating mode and db for your fasta?

**Comment by Anto007** [comment]
`--db 3` and `--mode 3`. The total run time was an impressive 3 hours.

**Comment by DominikBuchner** [comment]
This is really strange, I can confirm that with your .fasta I can also get results >85 > 94% but was not able to reproduce this with any other file! Can you tell me the operating system this file was produced on?

**Comment by Anto007** [comment]
@DominikBuchner It was generated on an Ubuntu 20.04 LTS OS after running dada2 and some final ASV filtering steps such as removal of NuMTs, non-Eukaryota ASVs and so on.

**Comment by DominikBuchner** [comment]
Hm okay, I officially don't get it. Let's see if I get a positive response from BOLD, so far, it does not seem to be fixed (except for Anto's file :D)

**Comment by Anto007** [comment]
Very odd..What if perhaps you made a new hybrid test fasta file containing, for example, 10 of my sequences and 10 of yours?

**Comment by DominikBuchner** [comment]
Maybe a good idea. Will test tomorrow.

**Comment by vanessamata** [comment]
interesting... I've tried re-running singles sequences on the website and I still get the same issue, no matches or only matches >94%. It's very odd that for a specific fasta files it does provide results >85%... I am very confused...! I haven't had any feedback from BOLD since I provided example sequences :(

**Comment by DominikBuchner** [comment]
So I got feedback from BOLD: They say the website is working as intended and there is no bug. I'm as confused as you are, but will perform further tests.
They will publish an API around January/February which will speed up the whole process once implemented into BOLDigger3. I'll keep you updated, the other bug I reported was resolved today, so no more "unavailable" process IDs. I believe that it has sth. to do with the formatting of Anto's file and while doing some bug-testing with the same data in different formats. Will keep you updated.

**Comment by vanessamata** [comment]
their website is working so well that I have been waiting for over 10 minutes for a single sequence with no luck... boldigger3 also seems to be stuck. Maybe their identification engine is down?

V4 seems to be working again fine though xD would it be difficult to update boldigger2 to use the new address of V4?

**Comment by DominikBuchner** [comment]
It appears to be the length of the sequence. Sequences shorter than 225 bp produce this bug. I reported it and wait for a response.

**Comment by Anto007** [comment]
Ohhh...weird but great to know that you were finally able to identify the bug here

**Comment by vanessamata** [comment]
@DominikBuchner I got the following answer from BOLD today:

"We had our team look into this and they concluded that the reason it doesn't show up is that it doesn't meet the other parameters for a valid match. Specifically, the parameter overlap is too low (query seq length - gaps) for certain match configurations even though results may reach 90%-93% identification.
We may re-examine these parameters in the future, but for now we suggest using v4 for your analysis in this case."

This is a big bummer for people working with degraded DNA (small sequences), like dietary analysis and environmental DNA. I know it's probably annoying, but can you update boldigger2 so it functions with v4?

**Comment by DominikBuchner** [comment]
Hi Vanessa, 
I'm not sure I got this correctly. If we find out h ow this overlap parameter works I can probably tweak it in a way that lower matches are accepted. I agree that this sucks, but backporting the code to v4 a) makes no sense, since it will be taken down soon b) does not work, since they shut down the API that I heavily used in BOLDigger2. 

So, to move forward: Let's find out how this overlap parameter works, then maybe we can tweak BOLDigger3. Will take a look into this!

**Comment by vanessamata** [comment]
Well, if you can figure out how to bypass that parameter, that would be great! :D I assumed you couldn't really tweak their algorithm...! But if they are using blast in the background, maybe that's possible...? I wish I could be more of help instead of crying for help eheh so thanks for your effort and nonetheless very useful non-bioinformatician skills ;)

**Comment by DominikBuchner** [comment]
At least it is quite easy to change the max hits returned and also go down with similarity values to 0.8. SO I guess I can also reverse engineer how to change the minimum overlap.

**Comment by vanessamata** [comment]
🤞🤞🤞

**Comment by DominikBuchner** [comment]
Okay, the overlap parameters can be changed, however I did not find out how this affects the results (it does change them, but I don't have sufficient time to do robust testing atm). I have to finish off a few other urgent things first, then I'll come back to this!

**Comment by vanessamata** [comment]
@DominikBuchner sorry for being a pain, buuut, any news on this? :) lot's of small projects stopped and waiting for proper IDs :(

**Comment by DominikBuchner** [comment]
Sorry, not yet :( And I probably won't manage to do it this year, since I need to finish of a lot of large projects :/ Will keep you updated.

**Comment by DominikBuchner** [comment]
Small update here: The limit will be reduced to 200 bp soon.

**Comment by DominikBuchner** [comment]
@vanessamata another small update: Changing the parameter does not change anything, as it seems to be hardcoded in the BOLD search itself. However, they plan to release an API for the search engine soon and I will see if I can work this into BOLDigger to fix this. Also the limit will be reduced to 200 bp, and might be reduced even further given enough community interest. I'll keep you posted.

**Comment by vanessamata** [comment]
@DominikBuchner that's annoying :( but if they reduce the limit to 200bp, that will already help me a lot, as I am mostly using Fwh2 (205bp). For people using Fwh1 or ANML, that won't be enough though. Hopefully you will be able to work around it with the API. Any idea when the 200bp limit will be implemented? I have datasets pilling up, waiting for this :))))

**Comment by DominikBuchner** [comment]
We are also almost exclusively using fwh2. From what I know, end of this
week :)

vanessamata ***@***.***> schrieb am Mo., 20. Jan. 2025, 12:24:

**Comment by vanessamata** [comment]
@DominikBuchner any news? I just tried a random fwh2 sequence (that happens to be a bat) on the website and still returns as no match because there are no matches above 94% :(

**Comment by OndroV** [comment]
Hello, 
my question is slightly off topic, but came to mind when thinking about fwhF2 and fragment length. The product of fwhF2 with the [promising](https://onlinelibrary.wiley.com/doi/full/10.1002/edn3.177) EPTDr2n is also only 142 bp. Is that primer not used anymore for water samples? Seems like people still prefer to get many nontarget reads rather than switching to that primer. We had a mixed experience: sequencing troubles hinting towards primer-dimers despite thorough purification, but eventually very good proportion of target reads. This issue with 200 bp threshold at BOLD is another con in this regard.
... If I can dare to ask for your current recommendations about this "minimized nontarget amplification" @DominikBuchner @TillMacher @VascoElbrecht 😃 

Cheers, fingers crossed and thank you all for the great tools!
Ondrej

**Comment by DominikBuchner** [comment]
> [@DominikBuchner](https://github.com/DominikBuchner) any news? I just tried a random fwh2 sequence (that happens to be a bat) on the website and it still returns as no match because there are no matches above 94% :(

Hi Vanessa, sorry for the late reply. We are on it.

**Comment by DominikBuchner** [comment]
> Hello, my question is slightly off topic, but came to mind when thinking about fwhF2 and fragment length. The product of fwhF2 with the [promising](https://onlinelibrary.wiley.com/doi/full/10.1002/edn3.177) EPTDr2n is also only 142 bp. Is that primer not used anymore for water samples? Seems like people still prefer to get many nontarget reads rather than switching to that primer. We had a mixed experience: sequencing troubles hinting towards primer-dimers despite thorough purification, but eventually very good proportion of target reads. This issue with 200 bp threshold at BOLD is another con in this regard. ... If I can dare to ask for your current recommendations about this "minimized nontarget amplification" [@DominikBuchner](https://github.com/DominikBuchner) [@TillMacher](https://github.com/TillMacher) [@VascoElbrecht](https://github.com/VascoElbrecht) 😃
> 
> Cheers, fingers crossed and thank you all for the great tools! Ondrej

This is a rather complicated question, I'll try to hide my disliking of eDNA as best as I can:

- We rarely use the primer in our routine programs but prefer PCR replication to pick up rare signals. We are currently performing 2 Extraction replicates + 4 PCR replicates for all our eDNA studies which helps a lot with rare signal detection.
- Regarding the primer dimers: I'm working on publishing my complete metabarcoding workflow on protocols.io. TLDL: Cleanup after the first PCR; Cleanup after the 2nd PCR; Reconditioning PCR of the Library --> no more primer dimers. I've sequenced over 50 librarys last year and this is not a problem in any of them.
- The primer does miss some important groups in freshwater monitoring. I anything, I'd use fwh2 + EPTD on top of that. This is a lot of PCRs and cleanups and I can see how people prefer not to go that route without a pipetting robot.
- The general problem with nontarget amplification is that you don't know when and how much of an effect you have to expect, since you'd have to know the composition of your sample prior to analysis. This is a general problem with eDNA and needs more research imho, so I do not have a good recommendation other than going for good old bulk samples or accepting this flaw of eDNA.

**Comment by vanessamata** [comment]
Thanks @DominikBuchner ! Those are great news :D do you know if it's working for any fragment length or just for sequences >200bp?

**Comment by DominikBuchner** [comment]
ItÄs lowered to 100 bp!
